compiler-core/src/analyse.rs

```diff
@@ -405,9 +405,16 @@ fn register_types_from_custom_type<'a>(
 
     hydrator.clear_ridgid_type_names();
 
+    // If the type comes from an internal module we change it's publicity to
+    // `Internal`.
+    let publicity = if config.is_internal_module(module) {
```

Comment:
Doesn't this now make private ones internal?

Reply:
Oh you're right! I have to restrict it to internal only if it's public. I'll add a test